webapi/src/main/scala/org/knora/webapi/slice/admin/api/service/ProjectsADMRestService.scala

```diff
@@ -232,13 +239,28 @@ final case class ProjectsADMRestServiceLive(
    *
    * @param id      the [[ProjectIdentifierADM]] of the project
    * @return
-   *     '''success''': the restricted view settings as [[ProjectRestrictedViewSettingsGetResponseADM]]
+   *     '''success''': the restricted view settings as [[ProjectRestrictedViewSettingsResponseADM]]
    *
    *     '''failure''': [[dsp.errors.NotFoundException]] when no project for the given [[ProjectIri]] can be found
    */
-  def getProjectRestrictedViewSettings(id: ProjectIdentifierADM): Task[ProjectRestrictedViewSettingsGetResponseADM] =
+  def getProjectRestrictedViewSettings(id: ProjectIdentifierADM): Task[ProjectRestrictedViewSettingsResponseADM] =
     responder.projectRestrictedViewSettingsGetRequestADM(id)
 
+  /**
+   * Sets project's restricted view settings.
+   *
+   * @param iri the project's IRI,
+   * @param user requesting user,
+   * @param size value to be set,
+   * @return [[ProjectRestrictedViewSettingsResponseADM]].
+   */
+  override def setProjectRestrictedViewSettings(
+    iri: Iri.ProjectIri,
+    user: UserADM,
+    size: RestrictedViewSize
+  ): Task[ProjectRestrictedViewSizeResponseADM] =
+    responder.setProjectRestrictedViewSettings(iri, user, size)
```

Comment:
Here we should check whether a project really exists with the given iri. Currently your code would simply persist a triple also if it is a valid `iri` but the project does not exist in the db.